Reworked frontend

NotJustJoe opened the pull request.

Commit:
reworked frontend and several types

Commit:
Merge remote-tracking branch 'origin' into staging

Commit:
updated missing runes, fixed movie loaded staying loaded when going out of view port

LaBoiteDeCarton commented on apps/front/src/lib/components/tadflix/movie-card/movie-card.svelte:
```diff
@@ -3,43 +3,52 @@
-			});
-	}
+
+	$effect(() => {
+		if (isVisible) {
+			isLoading = true;
+			loadBackdropImage(movieId, 'small')
+				.catch((error) => {
+					console.error('Error loading backdrop image:', error);
+					backgropImage = null;
+				})
+				.finally(() => {
+					isLoading = false;
+				});
+		}
+	});
```
Il semble ignorer le isVisible car le movieCard n'attend pas le scroll horizontal pour charger l'image.

LaBoiteDeCarton commented on apps/back/app/controllers/comments_controller.ts:
```diff
@@ -0,0 +1,12 @@
+import Comment from '#models/comment'
```
Une pr est en cours par @gmehdevi sur le comments, il faudrait supprimer la partie qui est hors scope pour eviter le conflits

LaBoiteDeCarton commented on apps/front/src/lib/components/tadflix/layout/navbar/Navbar.svelte:
```diff
@@ -0,0 +1,65 @@
+<script lang="ts">
```
La navbar sur netflix a un background degradé quand il est tout en haut (le scroll est a 0)
<img width="1404" alt="Capture d’écran 2025-07-04 à 10 26 49" src="https://github.com/user-attachments/assets/6c4d0d87-a8c6-44c4-9788-0479312fab61" />

Puis le background devient black une fois le scroll initié

LaBoiteDeCarton replied:
<img width="1380" alt="Capture d’écran 2025-07-04 à 10 26 44" src="https://github.com/user-attachments/assets/0b867e63-a984-4dcb-bd39-72c81586f94e" />

LaBoiteDeCarton commented on apps/front/src/lib/components/tadflix/movie-carousel/movie-carousel.svelte:
```diff
@@ -72,21 +78,21 @@
+			<CarouselItem class="basis-1/4 p-[3px] sm:basis-1/5 md:basis-1/6 lg:basis-1/7 xl:basis-1/8">
 				<div use:inView={(visible) => handleVisibility(index, visible)}>
 					<HoverCard openDelay={100} closeDelay={100}>
 						<HoverCardTrigger href="/movie/{movie.id}" target="_blank" rel="noreferrer noopener">
```
Tant qu'a refacto, ici le lien ouvre une nouvelle page au lieux de faire une redirection simple.

LaBoiteDeCarton commented on apps/front/src/lib/components/tadflix/movie-card/movie-card.svelte:
```diff
@@ -3,43 +3,52 @@
 	import { Skeleton } from '@/components/ui/skeleton';
```
Je sais d'où vient le probleme mais avant pour un movie dont on a pas le background image qui contient le nom du film avec la langue demandé on affiche le nom a la main et ça donne ça : 
<img width="329" alt="Capture d’écran 2025-07-04 à 10 33 44" src="https://github.com/user-attachments/assets/16b04369-5436-46e9-9f66-3ce38ae734bf" />
Après la refacto c'est peté : 
<img width="357" alt="Capture d’écran 2025-07-04 à 10 34 06" src="https://github.com/user-attachments/assets/be97cdcc-28b9-4878-a921-23068da066b5" />

LaBoiteDeCarton commented on apps/front/src/routes/(app)/(authed)/movie/[id]/+page.svelte:
```diff
@@ -0,0 +1,22 @@
+</script>
+
+<div class="flex flex-col overflow-y-scroll">
+	<MovieBanner movie={data.movie} />
```
Cette page c'est la page de la lecture de la vidéo et non celle de la Bannière. La bannière que tu as fait est a integré dans la HomePage et lui donnant un id de film (surement un film populaire du moment filtré selon le preference du user si connecté, en attendant tu peux lui filer le premier du TOP10 par exemple). Ce movie Banner est un peu le Header de la homepage.

Ne pas confondre avec la modal du film qui ressemble beaucoup au component movieDetails!

LaBoiteDeCarton commented on apps/front/src/lib/components/tadflix/movie-banner/MovieBanner.svelte:
```diff
@@ -0,0 +1,40 @@
+			<span class="md:max-w-[40%]">{movie.overview}</span>
+			<div class="relative flex w-full justify-between gap-2">
+				<div class="flex gap-2">
+					<Button class="bg-white">PLAY</Button>
+					<Button class="bg-[#6D6D6EB3]">MORE INFO</Button>
+				</div>
+				{#if movie.vote_average}
+					<div class="flex items-center gap-5">
```
Les boutons son pas ouf, il manque l'icone Play pour le bouton Play ('Lecture en Français)

LaBoiteDeCarton replied:
Fait. Il manque juste les langues

LaBoiteDeCarton commented on apps/front/src/routes/(app)/(authed)/movie/+page.svelte:
```diff
@@ -0,0 +1 @@
+<p>test</p>
```
Delete this file please

LaBoiteDeCarton reviewed (dismissed):
La refacto en Svelte 5 c'est top, les noms de variables sont + harmonisées ainsi que les states et les props.

Cependant quelques modification cruciales sont a faire nottament dans la position du component crée ainsi que les bugs ajoutés par la refacto.

Commit:
push fixed navbar, done 80% of comment feedback

LaBoiteDeCarton committed:
logo route added need to test

LaBoiteDeCarton committed:
Merge branch 'reworked-frontend' of github.com:asebrech/Hypertube into reworked-frontend

Commit:
Add logo support to MovieBanner and update API imports

- Enhanced MovieBanner component to optionally display a logo image.
- Updated API service imports to include BackDropImage type.
- Modified movie loading logic to fetch and pass logo image to MovieBanner.

LaBoiteDeCarton committed:
adding vidéo to movie banner. correcting link for movie preview

LaBoiteDeCarton committed:
button done

LaBoiteDeCarton committed:
adding langagues to button, fade away text when video ready until the end

LaBoiteDeCarton committed:
adding volume button to moviebanner, adding ratio to image for better affichage

LaBoiteDeCarton committed:
adding score for movie banner

LaBoiteDeCarton committed:
langage selector bg transparency

LaBoiteDeCarton committed:
replay vidéo movie banner when finished

LaBoiteDeCarton committed:
better fade effect on bottom (vidéo and backdrop)

LaBoiteDeCarton committed:
arranging percent of fader

LaBoiteDeCarton committed:
get rid of comment (il reste le component mais on le garde pour la trace

LaBoiteDeCarton committed:
correcting randomness + translate navbar

LaBoiteDeCarton committed:
tv shows added a la volée parce que c'etait easy

LaBoiteDeCarton committed:
delete console log in back

LaBoiteDeCarton committed:
,

LaBoiteDeCarton committed:
no charging, showing only when playing

LaBoiteDeCarton committed:
video logic inside movie card

LaBoiteDeCarton committed:
sound added to movie preview but opens movie. to correct

LaBoiteDeCarton commented:
Ça serait top de la faire passer et les Changes Request les mettre dans une issue pour avancer.

LaBoiteDeCarton committed:
requestChanges

LaBoiteDeCarton commented:
@asebrech

asebrech commented:
> @asebrech

J'allais le faire mais top merci !
(j'adore copilot)

asebrech reviewed (approved):
Incroyable c'est Netflix !!

J'ai quand meme ouvert cette issue chez eux : https://github.com/users/asebrech/projects/2?pane=issue&itemId=119497720&issue=asebrech%7CHypertube%7C70
